infra/Pos/Network/CLI.hs

```diff
@@ -91,47 +92,60 @@ defaultDnsDomains = DnsDomains [["todo.defaultDnsDomain.com"]]
 -------------------------------------------------------------------------------}
 
 -- | Interpreter for the network config opts
-intNetworkConfigOpts :: NetworkConfigOpts -> IO T.NetworkConfig
+intNetworkConfigOpts :: NetworkConfigOpts -> IO (T.NetworkConfig DHT.KademliaParams)
 intNetworkConfigOpts cfg@NetworkConfigOpts{..} = do
     parsedTopology <- case networkConfigOptsTopology of
                         Nothing -> return defaultTopology
                         Just fp -> readTopology fp
     ourTopology <- case parsedTopology of
       Y.TopologyStatic allStaticallyKnownPeers ->
-        fromPovOf cfg allStaticallyKnownPeers networkConfigOptsSelf
+        fromPovOf cfg allStaticallyKnownPeers networkConfigOptsSelf $ \nodeType peers ->
+          case nodeType of
+            T.NodeCore -> return (T.TopologyCore peers)
+            T.NodeRelay -> withKademliaParams cfg $ \kparams ->
+              return (T.TopologyRelay peers kparams)
+            -- This will never happen. Either our node name is not found in
+            -- the table, or it's a core or relay.
+            T.NodeEdge -> throwM NetworkConfigSelfEdge
       Y.TopologyBehindNAT dnsDomains ->
         return $ T.TopologyBehindNAT dnsDomains
-      Y.TopologyP2P -> return T.TopologyP2P
-      Y.TopologyTraditional -> return T.TopologyTraditional
-    mKademliaParams <- case networkConfigOptsKademlia of
-      Nothing -> return Nothing
-      Just fp -> do
-        kconf <- parseKademlia fp
-        kconf' <- either (throwM . DHT.MalformedDHTKey) return (DHT.fromYamlConfig kconf)
-        return $ Just kconf'
+      Y.TopologyP2P -> withKademliaParams cfg $ \kparams ->
+        return (T.TopologyP2P kparams)
+      Y.TopologyTraditional -> withKademliaParams cfg $ \kparams ->
+        return (T.TopologyTraditional kparams)
     return T.NetworkConfig {
         ncTopology    = ourTopology
-      , ncKademlia    = mKademliaParams
       , ncDefaultPort = networkConfigOptsPort
       , ncSelfName    = networkConfigOptsSelf
       }
 
+withKademliaParams :: NetworkConfigOpts
+                   -> (DHT.KademliaParams -> IO r)
+                   -> IO r
+withKademliaParams cfg k = case networkConfigOptsKademlia cfg of
+    Nothing -> throwM MissingKademliaConfig
```

Comment:
So does this address the TODO I saw elsewhere about needing two bits of config to be in sync for this. If so lets remove that TODO now.